CodingGuidelines.md

```diff
@@ -126,17 +152,29 @@ Always declare an access modifier for all fields, properties and methods.
 
 >All Unity API Methods should be `private` by default, unless you need to override them in a derived class. In this case `protected` should be used.
 
+>Fields should always be `private`, with `public` or `protected` property accessors.
```

Comment:
Does this imply that we will not allow readonly fields (essentially only allow get-only properties)? Feels a bit heavy handed.

Reply:
With the architecture of the framework, there should not be much need for readonly fields, especially as most things are made configurable or extendable.
But a lot of the guidance was formed based from issues found in the HTK and to avoid them in the future.

Reply:
Not sure read only fields apply here. I think those are fine to be public/protected as they're not being changed at runtime.

The principal goal here is ensuring proper encapsulation.